rest_framework_simplejwt/serializers.py

```diff
@@ -104,9 +104,25 @@ class TokenRefreshSerializer(serializers.Serializer):
     access = serializers.CharField(read_only=True)
     token_class = RefreshToken
 
+    default_error_messages = {
+        "no_active_account": _("No active account found for the given token.")
+    }
+
     def validate(self, attrs: Dict[str, Any]) -> Dict[str, str]:
         refresh = self.token_class(attrs["refresh"])
 
+        user_id = refresh.payload.get(api_settings.USER_ID_CLAIM, None)
+        if user_id and (
+            user := get_user_model().objects.get(
+                **{api_settings.USER_ID_FIELD: user_id}
+            )
+        ):
```

Comment:
Should this come with an else clause that also fails the validation? I don't think refresh should work for deleted users.

Reply:
Indeed it should not succeed.
Here get_user_model().objects.get(
                **{api_settings.USER_ID_FIELD: user_id}
            )
should raise a DoesNotExist exception if the user doesn't exist.

I can write a test for that as well.